packages/frontpage/app/(app)/layout.tsx

```diff
@@ -86,15 +97,42 @@ export default async function Layout({
 async function LoginOrLogout() {
   const session = await getSession();
   if (session) {
+    const did = await getDidFromHandleOrDid(session.user.name as string);
     return (
-      <form
-        action={async () => {
-          "use server";
-          await signOut();
-        }}
-      >
-        <button type="submit">Logout ({session.user.name})</button>
-      </form>
+      <DropdownMenu>
+        <DropdownMenuTrigger>
+          {did ? (
+            <UserAvatar did={did} size="smedium" />
+          ) : (
+            <span>{session.user.name}</span>
+          )}
+        </DropdownMenuTrigger>
+        <DropdownMenuContent className="w-56" side="bottom" align="end">
+          <form
+            action={async () => {
+              "use server";
+              await signOut();
+            }}
+          >
+            <DropdownMenuLabel>My Account</DropdownMenuLabel>
```

Comment:
Show the handle here?

Reply:
handles are too long imo and can be unwieldy on mobile.